docs/snapshotting/versioning.md

```diff
@@ -40,41 +60,51 @@ A microVM state file is further split into four different fields:
 The translation is done field by field - the common fields are copied from source to target, and the fields that are unique to the target are (de)serialized with their default values.
 
-The _semantic translator_ works on top of the structural one and its only concern is translating the semantics of the serialized/deserialized fields. 
+The _semantic translator_ is only concerned in translating the semantics of the serialized/deserialized fields. 
 
-The _structural translator_ is generated automatically through a procedural macro, and the _semantic translation methods_ have to be annotated in the structure by the user.
+The _field translator_ is generated automatically through a procedural macro, and the _semantic translation methods_ have to be annotated in the structure by the user.
 
 This block diagram illustrates the concept:
 
 ![Versionize](
 ../images/versionize.png?raw=true
 "Versionize layers")
 
-## Bincode encoding
-During research and prototyping we have considered multiple storage formats. The criteria used for comparing these were performance, size, Rust support, specification, versioning support, community popularity and tooling. Performance, size and Rust support are hard requirements while all others can be the subject of trade-offs. 
+## VM state encoding
+
+During research and prototyping we considered multiple storage formats. The criteria used for comparing these are: performance, size, rust support, specification, versioning support, community and tooling. Performance, size and Rust support are hard requirements while all others can be the subject of trade offs.
+More info about this comparison can be found here (https://github.com/firecracker-microvm/firecracker/blob/9d427b33d989c3225d874210f6c2849465941dc0/docs/snapshotting/design.md).
+
+Key benefits of using *bincode* versus everything else:
 
-More info about this comparison can be found [here](https://github.com/firecracker-microvm/firecracker/blob/9d427b33d989c3225d874210f6c2849465941dc0/docs/snapshotting/design.md#we-also-looked-at-these-other-options).
+* Minimal snapshot size overhead
+* Minimal CPU overhead
+* Simple implementation
 
-Key benefits of using `bincode` versus everything else are:
-- Minimal snapshot size overhead,
-- Minimal CPU overhead,
-- Simple implementation.
+The current implementation relies on the  Serde bincode (https://github.com/servo/bincode) for the signed and unsigned integer primitive types and provides a custom implementation for higher level primitives like vectors and strings. We have also added extra checks for variable field lengths. 
+
+Versionize is compatible to Serde with bincode backend: structures serialized with versionize at a specific version can be deserialized with Serde. Also structures serialized with serde can be deserialized with versionize.
```

Comment:
what does "specific version" mean? any version?

Reply:
A state structure changes over time, there are many versions of it. Serde doesn't work across versions, so it needs the specific structure layout -> specific version.

Reply:
I see